### PR TITLE
"Extract all links from a webpage" example

### LINKED ISSUE #110: Extract all links from a webpage
Using [`reqwest`](https://crates.io/crates/reqwest) to fetch the page + [`html5ever`](https://crates.io/crates/html5ever) to extract links.

### PR DESCRIPTION
**Pull request by sb89**
fixes #110 

---

- [x] ~Blocked on https://github.com/servo/string-cache/pull/187~
- [x] Blocked on https://github.com/utkarshkukreti/select.rs/pull/37
- [ ] Alternatively blocked on: https://github.com/brson/rust-skeptic/issues/18

### CODE REVIEW

File: src/net.md [hunk 1]
```diff
@@ -831,6 +832,57 @@ shows Listening on 127.0.0.1:11500, run
...
+
+# error_chain! {
+#    foreign_links {
+#        ReqError(reqwest::Error);
+#        IoError(std::io::Error);
+#    }
+# }
+
+fn run() -> Result<()> {
+    let mut res = reqwest::get("https://www.rust-lang.org/en-US/")?;
```
Comment (budziq) [inline src/net.md]: `Reqwest::Response` implements `io::Read` I would suggest using `Document::from_read` in such way you can drop all the intermediate "buffer" and "html" lines leaving just
```rust
    let res = reqwest::get("https://www.rust-lang.org/en-US/")?;
    let document = Document::from_read(res)?;
```

File: src/net.md [hunk 2]
```diff
@@ -831,6 +832,57 @@ shows Listening on 127.0.0.1:11500, run
 
 After sending data in telnet press `ctrl-]` and type `quit`.
 
+[ex-extract-links-webpage]: #ex-extract-links-webpage
+<a name="ex-extract-links-webpage"/>
+## Extract all links from a webpage
+
+[![reqwest-badge]][reqwest] [![select-badge]][select] [![cat-net-badge]][cat-net]
+
+```rust
```
Comment (budziq) [inline src/net.md]: Please change 
<pre>```rust</pre>
to
<pre>```rust,no_run</pre>

We do not want to make network requests in our CI environment

**Comment by budziq** [comment]
@sb89  Also the CI fails due to one of `select` dependencies (namely `servo/string-cache`) using `serde` 0.9 while rest of cookbook examples dependencies using `serde` 1.0.

The situation is already fixed upstream but servo related crates are currently coordinating move to `serde` 1.0 which should happen shortly https://github.com/servo/string-cache/pull/187.

File: src/intro.md [hunk 3]
```diff
@@ -172,6 +173,8 @@ Keep lines sorted.
 [regex]: https://docs.rs/regex/
 [reqwest-badge]: https://badge-cache.kominick.com/crates/v/reqwest.svg?label=reqwest
 [reqwest]: https://docs.rs/reqwest/
+[select-badge]: https://badge-cache.kominick.com/crates/v/reqwest.svg?label=select
```
Comment (budziq) [inline src/intro.md]: the badge points to reqwest instead of select

**Comment by budziq** [comment]
Nicely done!

As this will likely not get merged until select deps upversion to serde 1.0 (otherwise we have a build break on other examples) please add a comment to 
 #110 that this PR is waiting for deps upversion.

cc @dtolnay

**Comment by brson** [comment]
Another PR blocked by skeptic bug https://github.com/brson/rust-skeptic/issues/18

**Comment by brson** [comment]
Thanks @sb89 !

**Comment by budziq** [comment]
update:
the https://github.com/servo/string-cache/pull/187 has been merged.

now we wait on select.rs update to most recent html5ever
https://github.com/utkarshkukreti/select.rs/pull/37

**Comment by budziq** [comment]
Great! Thanks for sticking with it @sb89 ! 🎉

**Pull request merged**